### PR TITLE
Dynamic learning rate

### PR DESCRIPTION
**Pull request by George3d6**
* Set learning rate dynamically when we start training the `Neural` mixer
* Removed optimization via optuna from the neural mixer for now since the num hidden was always set to 1 and we are selecting the lr with this new method
* Added `stable` flag to models, if set to `False` then if they fail to train lightwood won't crash even in strict mode (unless all models fail to train)

### CODE REVIEW

**Comment by George3d6** [comment]
Benchmarks pending should be done in like 3-4 hours tops.

File: AI2.json [hunk 1]
```diff
@@ -1 +1 @@
...
...nition.timeseries_settings", "search_hyperparameters": true}}, {"module": "LightGBM", "args": {"stop_after": "$problem_definition.seconds_per_model", "target": "$target", "dtype_dict": "$dtype_dict", "input_cols": "$input_cols", "fit_on_dev": true}}, {"module": "Regression", "args": {"stop_after": "$problem_definition.seconds_per_model", "target_encoder": "$encoders[$target]"}}], "ensemble": {"module": "BestOf", "args": {"accuracy_functions": "$accuracy_functions", "target": "$target", "data": "test_data", "models": "$models"}}}, "problem_definition": {"target": "MEDV", "nfolds": 30, "pct_invalid": 1, "unbias_target": false, "seconds_per_model": 40, "seconds_per_encoder": 327, "time_aim": 200, "target_weights": null, "positive_domain": false, "fixed_confidence": null, "timeseries_settings": {"is_timeseries": false, "order_by": null, "window": null, "group_by": null, "use_previous_target": false, "nr_predictions": null, "historical_columns": null, "target_type": ""}, "anomaly_detection": true, "anomaly_error_rate": null, "anomaly_cooldown": 1, "ignore_features": [], "fit_on_validation": true, "strict_mode": true}, "identifiers": {}, "imports": [], "accuracy_functions": ["r2_score"]}
```
Comment (paxcema) [inline AI2.json:1]: This file should be removed, right?

File: lightwood/api/json_ai.py [hunk 2]
```diff
@@ -503,8 +504,10 @@ def code_from_json_ai(json_ai: JsonAI) -> str:
         model.fit(train_data)
         trained_models.append(model)
     except Exception as e:
-        if {json_ai.problem_definition.strict_mode} and 'regression' not in str(type(model)).lower():
+        log.warning(f'Excpetion: {{e}} when training model: {{model}}')
```
Comment (paxcema) [inline lightwood/api/json_ai.py:507]: Typo 

> "excpetion"

File: lightwood/model/neural.py [hunk 3]
```diff
@@ -80,34 +79,71 @@ def _select_criterion(self) -> torch.nn.Module:
 
         return criterion
 
-    def _select_optimizer(self, lr) -> Optimizer:
+    def _select_optimizer(self) -> Optimizer:
+        # ad_optim.Ranger
```
Comment (paxcema) [inline lightwood/model/neural.py:83]: leftover comments

File: lightwood/model/neural.py [hunk 4]
```diff
@@ -80,34 +79,71 @@ def _select_criterion(self) -> torch.nn.Module:
...
+                        scaler.scale(loss).backward()
+                        scaler.step(optimizer)
+                        scaler.update()
+                    else:
+                        loss.backward()
+                        optimizer.step()              
+                cum_loss += loss.item()
+
+                # Account for ranger lookahead update
+                print(batches)
```
Comment (paxcema) [inline lightwood/model/neural.py:128]: Leftover `print`

File: lightwood/model/neural.py [hunk 5]
```diff
@@ -80,34 +79,71 @@ def _select_criterion(self) -> torch.nn.Module:
...
+                if batches >= 1200:
```
Comment (paxcema) [inline lightwood/model/neural.py:129]: Shouldn't this be some function of `len(dl)`?

File: lightwood/model/neural.py [hunk 6]
```diff
@@ -118,8 +154,28 @@ def _max_fit(self, train_dl, dev_dl, criterion, optimizer, scaler, stop_after, r
 
         train_error = None
         for epoch in range(1, return_model_after + 1):
-            train_error = self._run_epoch(train_dl, criterion, optimizer, scaler)
-            #log.info(f'Train error: {round(train_error,3)}')
+            print(f'Epoch: {epoch}')
```
Comment (paxcema) [inline lightwood/model/neural.py:157]: Either `log.info` or remove `print`

File: lightwood/model/neural.py [hunk 7]
```diff
@@ -200,43 +259,55 @@ def objective(trial):
...
+        dev_ds_arr = ds_arr[int(len(ds_arr) * 0.9):]
 
-        net_kwargs = {'input_size': len(ds_arr[0][0][0]),
-                      'output_size': len(ds_arr[0][0][1]),
-                      'num_hidden': self.num_hidden,
-                      'dropout': 0}
+        con_train_ds = ConcatedEncodedDs(train_ds_arr)
+        con_test_ds = ConcatedEncodedDs(dev_ds_arr)
+        self.batch_size = 200
+        # self.batch_size = min(200, int(len(con_train_ds) / 50))
```
Comment (paxcema) [inline lightwood/model/neural.py:281]: Why not `min(200, int(len(con_train_ds) / 50))`?
